feat(Authoring): Allow expanding and collapsing lessons

geoffreykwan opened the pull request:
## Changes
- Added expand/collapse button to lessons
- Added border around lessons
- Added expand all button
- Added collapse all button

## Test
Make sure clicking
- the expand lesson button, shows the steps in the lesson
- the collapse lesson button, hides the steps in the lesson
- on a lesson, opens the lesson
- expand all, expands all lessons
- collapse all, collapses all lessons

Closes #1621

geoffreykwan committed:
feat(Authoring): Allow expanding and collapsing lessons

#1621

hirokiterashima commented on src/assets/wise5/authoringTool/project-authoring/project-authoring.component.ts:
```diff
@@ -159,4 +177,38 @@ export class ProjectAuthoringComponent {
+    });
+  }
+
+  protected expandedChangeEvent(event: ExpandEvent): void {
```
Rename function to something more apt, like "onExpandChanged" or "handleExpandChanged" or "expandElement"?

hirokiterashima commented on src/assets/wise5/authoringTool/project-authoring/project-authoring.component.ts:
```diff
@@ -8,18 +8,26 @@ import { temporarilyHighlightElement } from '../../common/dom/dom';
+  });
+  protected isAllLessonsExpanded: Signal<boolean> = computed(() => {
+    return this.isAllLessonsExpandedValue(true);
+  });
```
Remove return statement and surrounding brackets?

Also- should they be named allLessonsCollapsed and allLessonsExpanded because they're variables instead of functions?

hirokiterashima reviewed (changes requested):
The new UI looks great. It adds more organization and makes it easy to understand the structure of the unit better.

Functionality looks good for the most part. Here are some behavioral changes that we discussed today:
1. If a lesson is empty, hide its expand/collapse button
2. Collapsing a lesson should unselect any step(s) that are in that lesson

Also, I added few code improvement suggestions inline.

Is the tsconfig.json change necessary for this PR? What is it doing?

We don't have to address this in this PR, but something to think about for the future- I wonder if we should make ```lessonIdToExpanded``` into a shared state in the TeacherProjectService instead of passing the expanded field to the components? This might make the code simpler because there'll be less data passing between the components.

geoffreykwan committed:
fix(Authoring): Unselect step when parent lesson is collapsed

Also change some variable names and hide the expand/collapse button if a
lesson has no steps.

#1621

geoffreykwan commented:
Ok, I've made all the changes. The change to tsconfig.json is required in order to import the demo project json file for the tests.

https://github.com/WISE-Community/WISE-Client/blob/3e1c4f2ec2346b76feccd0d348616dd386e5bed3/src/assets/wise5/authoringTool/project-authoring/project-authoring.component.spec.ts#L23

hirokiterashima reviewed (approved):
LGTM

geoffreykwan commented:
:tada: This PR is included in version 5.132.0 :tada:

The release is available on [GitHub release](https://github.com/WISE-Community/WISE-Client/releases/tag/v5.132.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
